### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by NikitaBukhta**
POSIX library created: there are wrappings for all function we need. This library is contained in POSIX directory

Server class was added in directory server. There is server application too.

The file "main.cpp" was deleted, because that was a temp file.

UPD:
Client class was added in directory client. There is client application too.

UPD2:
Keepalive connection added on the server

UPD3:
Added server-side detection of client disconnection from the server.
Added keepalive for the client and for the server.
Added interactive and single message modes for the client

UPD4:
deleted using POSIX::_select() function
fixed bugs with sending messages

### CODE REVIEW

File: POSIX/include/posixError.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    {
+    private:
+        char *errorMessage;
+
+    public:
+        PosixError(const char *msg);
+        
+        ~PosixError(void);
+
+        virtual const char* what(void) const noexcept override;
```
Comment (anantonyk) [inline POSIX/include/posixError.h:17]: Why do you use all of this specifiers?

File: POSIX/include/posixError.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#define POSIX_ERROR_H
+
+#include <exception>
+
+namespace POSIX
+{
+    class PosixError : public std::exception
+    {
+    private:
+        char *errorMessage;
```
Comment (anantonyk) [inline POSIX/include/posixError.h]: use std::string

File: client/client-app.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        printf("Waiting for your message: ");
+        std::getline(std::cin, buffer);     // read full line;
+        if (buffer == "\n" || buffer == "") 
+            break;
+
+        singleMessageMode(client, buffer);
+    }
+}
+
+int main(int argc, char **argv)
```
Comment (anantonyk) [inline client/client-app.cpp:15]: write this on the begin of the file and put functions below

File: client/include/client.h [hunk 4]
```diff
@@ -0,0 +1,57 @@
+#ifndef CLIENT_H
+#define CLIENT_H
+
+#include <netinet/in.h>
+#include <string>
+
+namespace cs    // client-server
```
Comment (anantonyk) [inline client/include/client.h]: hmm. maybe two namespaces would be better?

File: server/include/connectionError.h [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+#define CONNECTION_ERROR_H
+
+#include <exception>
+
+namespace cs
+{
+    class ConnectionError : std::exception
+    {
+    private:
+        char *errorMessage;
```
Comment (anantonyk) [inline server/include/connectionError.h]: std::string

Comment (NikitaBukhta) [reply server/include/connectionError.h]: Done

File: client/include/client.h (hunk 4 above)
Comment (NikitaBukhta) [reply client/include/client.h]: Done

File: client/client-app.cpp (hunk 3 above)
Comment (NikitaBukhta) [reply client/client-app.cpp:15]: Done

File: POSIX/include/posixError.h (hunk 2 above)
Comment (NikitaBukhta) [reply POSIX/include/posixError.h]: Done

**Review by anantonyk — APPROVED** [review]

**Pull request merged**